packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2821,6 +2822,8 @@ enum _RouteLifecycle {
   removing, // we are waiting for subsequent routes to be done animating, then will switch to dispose
   // routes that are completely removed from the navigator and overlay.
   dispose, // we will dispose the route momentarily
+  disposing, // The entry is waiting for its subtree to disposed first. It should
+             // be in stored in the _entryWaitingForSubTreeDisposal
```

Comment:
subtree = widget subtree? Maybe a little clearer:

"The entry is waiting for its widget subtree to be disposed first. It is stored in _entryWaitingForSubTreeDisposal while awaiting that."